cmd/register/oidcconfig/cmd.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oidcconfig
+
+import (
+	// nolint:gosec
+
+	//#nosec GSC-G505 -- Import blacklist: crypto/sha1
+
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/briandowns/spinner"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/cmd/create/oidcprovider"
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/helper"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	installerRoleArn string
+	issuerUrl        string
+	secretArn        string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "oidc-config",
+	Aliases: []string{"oidcconfig", "oidcconfig"},
+	Short:   "Registers unmanaged OIDC config with Openshift Clusters Manager.",
+	Long:    "Registers unmanaged OIDC config with Openshift Clusters Manager.",
+	Example: `  # Register OIDC config
+	rosa create oidc-config`,
+	Run: run,
+}
+
+const (
+	installerRoleArnFlag = "installer-role-arn"
+	issuerUrlFlag        = "issuer-url"
+	secretArnFlag        = "secret-arn"
+
+	secretsManagerService = "secretsmanager"
+
+	minorVersionForGetSecret  = "4.12"
```

Comment:
NP
Can we extract the constants to a common file and reuse them in `create/oidcconfig/cmd`?